lab_1_classify_by_unigrams/main.py

```diff
@@ -11,15 +11,34 @@ def tokenize(text: str) -> list[str] | None:
     :param text: a text
     :return: a list of lower-cased tokens without punctuation
     """
-
+    if not isinstance(text, str):
+        return None
+    tokens = []
+    text = text.lower()
+    for i in text:
+        if i.isalpha():
+            tokens += i
+    return tokens
 
 def calculate_frequencies(tokens: list[str] | None) -> dict[str, float] | None:
     """
     Calculates frequencies of given tokens
     :param tokens: a list of tokens
     :return: a dictionary with frequencies
     """
-
+    if not isinstance(tokens, list):
+        return None
+    for i in tokens:
+        if not isinstance(i, str):
+            return None
+    dictionary = {}
+    length = len(tokens)
+    for i in tokens:
+        if i in dictionary:
+            dictionary[i] += 1/length
+        else:
```

Comment:
Can you think of a way to avoid this `else` here?

Reply:
Done